### DISCUSSION TITLE
[Bug]: Ordering by field in Datasource doesnt let you Sort with it

### DISCUSSION THREAD

**Discussion by BenOussama180** [Power-Components/livewire-powergrid#780] [Q&A]
### Have you searched through other issues to see if your problem is already reported or has been fixed?

No

### Did you read the documentation?

Yes, I did not find it.

### Have you tried to publish the views?

Yes - I didn't work.

### Is there an error in the console?

No

### PowerGrid

3.5.13

### Laravel

9.x

### Livewire

v2.10.5

### Alpine JS

3

### Theme

Tailwind 3.x, Tailwind 3.x with tailwind/forms

### Describe the bug.

when you order with a field in datasource for example ->orderBy('price', 'asc') u can not sort with it in the columns() function with the ->sortable() , the sort doesnt change

### To Reproduce...

data source: `
return Link::whereShowToCustomers(1)
            ->with('orders', 'orders.projects')
            ->orderBy('price', 'asc')
            ->groupBy('site');`
column : `
Column::add()
                ->title('Price')
                ->field('price_formatted', 'price')
                ->searchable()
                ->sortable()
                ->makeInputRange()
                ->bodyAttribute('text-center', '')
                ->headerAttribute('text-center', ''),`

### Extra information

```markdown
<?php
 //...
```

**Comment by luanfreitasdev** [comment]
I think you shouldn't use orderBy in datasource, try using $sortField and $sortDirection properties

https://livewire-powergrid.com/table/component-settings.html#sort-by-field-and-direction

**Comment by BenOussama180** [comment]
but i have to sort the data before grouping, so it'll only take the cheapest link out of duplicates by site , 
any work around ? @luanfreitasdev  thanks in advanced

**Comment by luanfreitasdev** [comment]
If groupBy is inside the powergrid component would it work?

**Comment by BenOussama180** [comment]
how can put the  groupby inside powergrid component? @luanfreitasdev

**Comment by luanfreitasdev** [comment]
> how can put the groupby inside powergrid component? @luanfreitasdev

after orderBy
https://github.com/Power-Components/livewire-powergrid/blob/main/src/PowerGridComponent.php#L260

**Comment by BenOussama180** [comment]
hmm i will try that! thank you

**Comment by luanfreitasdev** [comment]
Hello!. I'll move this up for discussion as it's not a bug initially. We can see this as an improvement.
